### DISCUSSION TITLE
멀티플레이 씬 이동 시 질문

### DISCUSSION THREAD

**Discussion by bonokbs** [naverz/zepeto-studio-kor#712] [5. Q&A]
안녕하세요. 멀티플레이 신 이동을 하는 코드를 만들고 있습니다.
![image](https://user-images.githubusercontent.com/32977809/167281600-9d6465ed-242d-49d4-8ec7-ba4519884454.png)

![image](https://user-images.githubusercontent.com/32977809/167281640-f9119c2c-0441-48c6-aa0d-d8999a79cfe8.png)

![image](https://user-images.githubusercontent.com/32977809/167281643-07064d48-a0b6-4db3-8cee-66b972cdcf8d.png)

Capsule 콜라이더에 닿았을시 -> GameB 씬으로 이동하라고 만들었습니다.
클라이언트 스타터, 서버는 제페토 공식 맵에서 소스코드 불러와 사용했습니다.
https://github.com/naverz/zepeto-multiplay-example

싱글플레이시에는 문제 없는데 멀티플레이시 이런 에러가 뜨는데
![image](https://user-images.githubusercontent.com/32977809/167281680-0dfdb67e-e0f2-4d88-9bfe-b37c0298d35e.png)

신 이동시  WorldMultiplay, ClientStarter 도 삭제하고 다시 불러와서 오류가 나는거 같습니다.

![image](https://user-images.githubusercontent.com/32977809/167281985-9ed24faa-9cab-4621-bdd1-c7a2f04c4ca1.png)

DontDestroyOnLoad 로 가야하는지...(타입스크립트에서 무엇을 임포트해야하는지 모르겠습니다.) 
거의 15시간동안 삽질하고 있는데 감을 못잡고 있습니다.

멀티플레이 신 이동하는 기능을 구현하고 싶은데 어떻게 만들어야 할지 모르겠습니다.

감사합니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, DontDestroyOnLoad 는 아래와 같이 사용할 수 있습니다.

import { Object } from 'UnityEngine'
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class DontDestoryObject extends ZepetoScriptBehaviour {

    Start() {    
        Object.DontDestroyOnLoad(this.gameObject);
    }

}

**Comment by smilejsu82** [comment]
> 안녕하세요. 멀티플레이 신 이동을 하는 코드를 만들고 있습니다.
> 
> ![image](https://user-images.githubusercontent.com/32977809/167281600-9d6465ed-242d-49d4-8ec7-ba4519884454.png)
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/32977809/167281640-f9119c2c-0441-48c6-aa0d-d8999a79cfe8.png)
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/32977809/167281643-07064d48-a0b6-4db3-8cee-66b972cdcf8d.png)
> 
> 
> 
> Capsule 콜라이더에 닿았을시 -> GameB 씬으로 이동하라고 만들었습니다.
> 
> 클라이언트 스타터, 서버는 제페토 공식 맵에서 소스코드 불러와 사용했습니다.
> 
> https://github.com/naverz/zepeto-multiplay-example
> 
> 
> 
> 싱글플레이시에는 문제 없는데 멀티플레이시 이런 에러가 뜨는데
> 
> ![image](https://user-images.githubusercontent.com/32977809/167281680-0dfdb67e-e0f2-4d88-9bfe-b37c0298d35e.png)
> 
> 
> 
> 신 이동시  WorldMultiplay, ClientStarter 도 삭제하고 다시 불러와서 오류가 나는거 같습니다.
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/32977809/167281985-9ed24faa-9cab-4621-bdd1-c7a2f04c4ca1.png)
> 
> 
> 
> DontDestroyOnLoad 로 가야하는지...(타입스크립트에서 무엇을 임포트해야하는지 모르겠습니다.) 
> 
> 거의 15시간동안 삽질하고 있는데 감을 못잡고 있습니다.
> 
> 
> 
> 멀티플레이 신 이동하는 기능을 구현하고 싶은데 어떻게 만들어야 할지 모르겠습니다.
> 
> 
> 
> 감사합니다.
> 
> 

#738을 확인해주세요^^
